src/events/event-bus.js

```diff
@@ -0,0 +1,56 @@
+import { E, wrapError } from './replayemitter'
+import * as C from '../utils/consts'
+
+function initBus (size) {
+  if (!size) {
+    size = 5
+  }
+  return new E(size)
+}
+
+function isNewEmitter (bus) {
+  return typeof bus.emitError === 'function' && typeof bus.encodeEmitError === 'function'
+}
+
+function extendBusIfNeeded (bus) {
+  if (isNewEmitter(bus)) {
+    return
+  }
+
+  bus.emitError = function (name, message, e = {}) {
+    const wrappedError = wrapError(name, message, e)
+    bus.emit(C.ERRORS_PREFIX, wrappedError)
+  }
+
+  bus.encodeEmitError = function (name, exception) {
+    bus.emitError(name, exception.message, exception)
+  }
+}
```

Comment:
Maybe we can rename the two to have less cryptic names. For example `emitError` to `emitErrorWithMessage` and `encodeEmitError` to `emitErrorExtractMessage`.

Reply:
+1 for  `emitError ` --> `emitErrorWithMessage` 
how about leaving `emitError` for `encodeEmitError ` ?

Reply:
I will leave it as it is for the old replayEmitter in case customers are emitting errors that way

Reply:
done